superx/information_extractor.py

```diff
@@ -22,133 +22,139 @@ class info_extractor:
-            # corrects the urls
-            if self.super_name == 'mega':
-                zip_link = self.url_list[0] + '/' + zip_link
-            elif self.super_name == 'victory':
-                zip_link = 'http://matrixcatalog.co.il/' + zip_link
-
-            request = requests.get(zip_link)
-            content = request.content
-            xml_file = gzip.decompress(content).decode('utf-8')
-            # parses the xml document into a tree
-            tree = ET.fromstring(xml_file)
-            # gets the branch/store ID of current xml file
-            branch_id = self.get_branch_id(tree.getchildren())
-            # gets child containing item information
-            items = tree.getchildren()[-1]
-            self.extract_information(items, branch_id)
-
-    def extract_information(self, items, current_branch_id):
+            # fix zip link url if neccessary
+            if not self.current_super['zip_link_prefix'] is None:
+                zip_link = self.current_super['zip_link_prefix'] + zip_link
+
+            try:
+                request = requests.get(zip_link)
+                content = request.content
+            except ConnectionError:
+                continue
+            else:
+                xml_file = gzip.decompress(content).decode('utf-8')
+                # parses the xml document into a tree
+                tree = ET.fromstring(xml_file)
+                # gets child containing item information
+                items = tree.getchildren()[-1]
+                self.extract_information(items)
+
+    def extract_information(self, items):
         """
         This method iterates over all items in the supermarket and extracts the relevant data
         The data is then committed into the relevant table in the data base
         :param items: The child of the parsed xml tree containing all item information
         :param current_branch_id: id of current branch
         """
-        products_list = []
-        branch_price_list = []
-        item = 'Item'
-        bIsWeighted = 'bIsWeighted'
+        item_attr_name = self.current_super['item_attr_name']
+        bIsWeighted = self.current_super['is_weighted_attr_name']
 
-        # different supermarkets use different titles
-        if self.super_name == 'victory':
-            item = 'Product'
-            bIsWeighted = 'BisWeighted'
-
-        for item in items.findall(item):
-            code = int(item.find('ItemCode').text)
+        for item in items.findall(item_attr_name):
+            item_code = int(item.find('ItemCode').text)
             name = item.find('ItemName').text
-            quantity = item.find('Quantity').text
-            price = item.find('ItemPrice').text
+            # exclude unwanted names from DB
+            for exclude in self.exclude_names:
+                if exclude in name:
+                    continue
+
+            quantity = float(item.find('Quantity').text)
+            price = float(item.find('ItemPrice').text)
```

Comment:
Please see the difference between float and decimal here:
https://docs.python.org/2/library/decimal.html

While float is faster than decimal, it is also not going to give you the exact number you want. Speed is not important in this context, as it is not the bottleneck.